app/models/concerns/season_phase_switcher.rb

```diff
@@ -0,0 +1,50 @@
+module SeasonPhaseSwitcher
```

Comment:
I would make this a class. Extracting instance methods into a module which is only used by that one class gains little in terms of true separation. On the contrary: it adds cognitive load since the functionality is not stated explicitly anymore. 

How about a class nested in the `Season` namespace?
```ruby
# app/models/season/phase_switcher.rb
class Season::PhaseSwitcher
  # …
end
```

The class can operate on `Season.current` directly.